### ISSUE TITLE
Several Weapons Have Stat Values of 0

### ISSUE THREAD

**Issue by Xalavar** [Bungie-net/api#773]
At first, I figured it was only Horror's Least (hash 1071542914) that had values of 0 for Impact, Attack, and Power until I noticed that Trust (hash 1833744687) also had 0 Impact, Attack, and Power, which leads me to believe that there might be more weapons following this sort of trend. 

I checked both the stats and investmentStats for both weapons, and the values were the same for the aforementioned stats.

**Comment by vthornheart-bng** [comment]
Is this coming down in live data, or in the definitions?

**Comment by LordNazo** [comment]
In the definitions.

**Comment by vthornheart-bng** [comment]
Ahh, yes.  This is a known issue, I'll have to look for the related bug when I get time.  A fundamental problem we run into on BNet is that we can't necessarily - at least at the moment - get the true stats at which all items can spawn.  The reason is extremely complicated, but my guess is that the heuristics we currently use are failing for these particular items.

I'll mark this as a bug for now, and loop back later when I get the chance to see if the old version of this bug still exists!

**Comment by vthornheart-bng** [comment]
I would say though that whatever impact, attack, power, defense, and speed stats you get from the *definitions* specifically should always be taken with a grain of salt.  I almost took them out before D2 went live because of the issues we were having with sourcing of this data, but I left it in for those who might still find the data useful.  The live data - actual instances of items - will always be the only thing that gives canonical truth for these stats because of the fact that we don't have a provably correct and programmatic way to verify the levels at which an item can spawn.

**Comment by ckhicks** [comment]
FWIW, I've even seen this in the Companion app, if it's coming from a source like Collections. I'm guessing that is because the stats could be affected by perks but it threw me for a moment to see a sniper rifle with a zero Impact rating. 😆

**Comment by LordNazo** [comment]
Actually, the only perks that can modify Impact are found in weapons with Charge Time stats in-game, such as Fusion Rifles and Linear Fusion Rifles.

**Comment by LordNazo** [comment]
SUROS Regime affects damage, but I don't believe its perks you're most likely referring to directly affect the Impact or Rate of Fire stats in-game the same way a perk that displays a value change in said stats would.

**Comment by floatingatoll** [comment]
I would not even attempt to hazard a guess in either scenario, whether
SUROS or LFRs, because Bungie has a history of using what I like to call
"hacks" to implement weapons using pieces of scripts from multiple other
kinds of weapons.

My point is: While the only perk _descriptions_ in-game that state that
modify Impact in a manner visible in the bar charts for a weapon are well
known to us players, that is no guarantee at all that those are the 'only'
perks. As players, we cannot be authoritative whatsoever here, due to
Bungie's frequent (and enjoyable - Coldheart rocks!) use of weapon script
customization as opposed to simple fixed constants.

**Comment by ckhicks** [comment]
Yeah, my point was more that I'd seen these inconsistencies out in the wild, not to find a specific case. Sorry for the confusion.

**Comment by LordNazo** [comment]
That makes sense. However, I still think Impact is too important of a stat to throw out, though, even with all of these hidden "hacks" that affect certain weapons. The majority of other weapons still have correct Impact values, so I'm glad those are still in.

**Comment by ckhicks** [comment]
Actually...I was correct, it was Impact that was bugged in the Companion app: https://imgur.com/a/ltl9sRk

**Comment by vthornheart-bng** [comment]
Aye, my personal inclination is to keep in at least the stats that I'm not being explicitly told to remove, even though all of them can potentially be modified by scripts and other game-related features that are beyond our control or ability to calculate.

This particular item is an interesting one though, I'd not seen an item before where impact was being treated in this way.  I will have to look into it when I get the time.

**Comment by floatingatoll** [comment]
To ask a question that's missing an answer here:

What 'value' does knowing Impact provide you, such that not being able to
see it hinders you enjoyment? You've indicated _that_ it's important, but
you haven't indicated _how_ it's important, which is making it difficult to
follow.

Is it being used to calculate _estimated_ (rather than _actual_) TTK? Is it
being used to rank weapons in order from highest to lowest damage per shot?

**Comment by LordNazo** [comment]
It usually goes hand-in-hand with the other available stats such as fire rate (RPM, Charge Time, etc.), Stability, Handling, etc. in determining which ones might be more effective to use. 

I personally use Impact to determine the effective combination of weapons to use in the Crucible. In some matches, I would prefer to use Erentil FR4, a high impact fusion rifle, with Liquid Coils (which maximizes Impact at the cost of charge time) due to how much more effective it is with increased damage, but sometimes I need to use other fusion rifles that fire faster, and Impact is usually an important factor for me as I sometimes "pre-charge" the fusion rifle and need to be certain that it can do enough damage. In the event that I decide to play more aggressively and fire from longer ranges, having a high impact fusion rifle is helpful when considering the falloff damage and how many of the projectiles would actually need to hit an enemy to kill them. Even if the Impact does not precisely reflect TTK across other weapons, it's better than not seeing it at all.
